interp/constrexpr_ops.ml

```diff
@@ -54,7 +54,7 @@ let names_of_local_binders bl =
    are considered different here. *)
 
 let prim_token_eq t1 t2 = match t1, t2 with
-| Numeral (s1,n1), Numeral (s2,n2) -> String.equal n1 n2 && s1 == s2
+| Numeral (s1,n1), Numeral (s2,n2) -> s1 == s2 && NumTok.equal n1 n2
```

Comment:
While you're at it, use `Int.equal`.

Reply:
This is not an int but a sum type with only constant constructors. I replaced it with a proper pattern matching (https://github.com/coq/coq/pull/8764/commits/234572e65fd0039e08c9b9eec34f3124b7c2e557).